### PR TITLE
Feature/20230602 julian

### PR DESCRIPTION
**Pull request by Julian0701**
### DEVELOP
- [x] DatePicker
- [x] OpenSubTab
- [x] ReceiptSection
- [x] ReceiptItem
- [x] MarketContext

### CHECK LIST
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component:  0
- new loop: 2
  - src/components/receipt_section/receipt_section.tsx line20
  - src/components/receipt_section/receipt_section.tsx line51
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/receipt_section/receipt_section.tsx [hunk 1]
```diff
@@ -54,37 +54,59 @@ const ReceiptSection = () => {
         return result;
       });
       setFilteredReceipts(searchResult);
-    } else if (filteredTradingType === '' && searches === '') {
-      setFilteredReceipts(listHistories);
-    } else if (filteredTradingType === OrderType.DEPOSIT) {
+    } else if (filteredTradingType !== '') {
```
Comment (Julian0701) [inline src/components/receipt_section/receipt_section.tsx:57]: 改成連續用 filter

File: src/components/open_sub_tab/open_sub_tab.tsx [hunk 2]
```diff
@@ -6,77 +6,103 @@ import {MarketContext} from '../../contexts/market_context';
...
+        const tickerPrice = marketCtx.availableTickers[cfd.targetAsset]?.price;
+        const spread = getTickerSpread(cfd.targetAsset);
+        /**
+         * Info: (20230428 - Shirley)
+         * without `positionLineGraph`, use market price to calculate
+         * without `market price`, use the open price of the CFD to get PNL === 0 and display `--`
+         * (OpenPositionItem & UpdateFormModal)
+         */
+        const currentPrice =
+          positionLineGraph.length > 0
```
Comment (Julian0701) [inline src/components/open_sub_tab/open_sub_tab.tsx:75]: tickerPrice 優先

File: src/components/open_sub_tab_mobile/open_sub_tab_mobile.tsx [hunk 3]
```diff
@@ -41,31 +54,52 @@ const OpenSubTabMobile = () => {
...
+        const spread = getTickerSpread(cfd.targetAsset);
+
+        /**
+         * Info: (20230428 - Shirley)
+         * without `positionLineGraph`, use market price to calculate
+         * without `market price`, use the open price of the CFD to get PNL === 0 and display `--`
+         * (OpenPositionItem & UpdateFormModal)
+         */
+        const currentPrice =
+          positionLineGraph.length > 0
```
Comment (Julian0701) [inline src/components/open_sub_tab_mobile/open_sub_tab_mobile.tsx:76]: tickerPrice 優先

**Review by arealclimber — APPROVED** [review]
- no bug found

**Review by TzuHanLiang — APPROVED** [review]
- no bug found

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**